amethyst_ui/src/button/builder.rs

```diff
@@ -362,15 +364,29 @@ impl<G: PartialEq + Send + Sync + 'static> UiButtonBuilder<G> {
             )
             .expect("Unreachable: Inserting newly created entity");
 
-        res.button
+        res.button_text_color
             .insert(
                 image_entity,
-                UiButton {
+                UiButtonTextColor {
                     text_color: self.text_color,
                 },
             )
             .expect("Unreachable: Inserting newly created entity");
 
+        let widget_id = if self.name.len() > 0 {
```

Comment:
So in this case, does every entity _need_ an `EntityId` that's user defined? Or can the user choose to omit it, in which case it will be auto generated? For now, I'll fall back on using the entity handle as the identifier, we can replace that with the `EntityId` component as soon as that's done.